CppCoreGuidelines.md

```diff
@@ -6539,6 +6540,40 @@ That tends to work better than "cleverness" for non-specialists.
 
 * Flag throwing `hash`es.
 
+### <a name="Rc-memset"></a>C.90: Rely on constructors and assignment operators, not `memset` and `memcpy`
+
+##### Reason
+
+The standard C++ mechanism to construct a an instance of a type is to call its constructor. As specified in [C.41: A constructor should create a fully initialized object](#Rc-complete). No additional initialization, such as by `memcpy`, should be required.
+A type will provide a copy constructor and/or copy assignment operator to appropriately make a copy of the class, preserving the type's invariants.  Using memcpy to copy a non-trivially copyable type can result in slicing, or data corruption. 
+
+##### Example, bad
+
+    struct base
+    {
+        virtual void update() = 0;
+    };
+
+    struct derived : public base
+    {
+        void update() override {}
+    };
+
+    // goodbye v-tables
+    void f(derived& a, derived& b)
+    {
+        memset(&a, 0, sizeof(derived));
+        memcpy(&a, &b, sizeof(derived));
```

Comment:
You are first zeroing out a and then copying over the bits from b.  Is that intentional?
Also, while that it's certainly UB, I would expect a and b to have a valid v-table after this on most implementations.

Reply:
Updated to split the operations and rewrote the example with an example motivated by an issue we saw in my product.